base/interactiveutil.jl

```diff
@@ -64,7 +64,7 @@ function edit(path::AbstractString, line::Integer=0)
     nothing
 end
 
-function edit(m::Method)
+function edit(m::TypeMapEntry)
```

Comment:
I think the fastest way to fix this is to put a `sig` field back in `Method`, then always return arrays of `Method`s from these functions. Then the cache representation would be completely hidden and I believe we would be backwards-compatible.

Of course, the only small problem then is the redundant `sig` field between Method and TypeMapEntry. To me it looks possible to have TypeMap look inside Method for this when necessary --- especially since TypeMapEntry actually has 2 sig fields, which I suspect could be reduced to 1.